packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -1897,5 +1952,56 @@ export function writeClientRenderBoundaryInstruction(
   }
 
   writeChunk(destination, boundaryID);
+  if (errorHash || errorMessage || errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    if (errorHash)
+      writeChunk(
+        destination,
+        stringToChunk(escapeJSStringsForInstructionScripts(errorHash)),
+      );
+  }
+  if (errorMessage || errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    if (errorMessage)
+      writeChunk(
+        destination,
+        stringToChunk(escapeJSStringsForInstructionScripts(errorMessage)),
+      );
+  }
+  if (errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    writeChunk(
+      destination,
+      stringToChunk(escapeJSStringsForInstructionScripts(errorComponentStack)),
+    );
+  }
   return writeChunkAndReturn(destination, clientRenderScript2);
 }
+
+const regexForJSStringsInScripts = /[<\'\"\`\n\u2028\u2029]/g;
```

Comment:
Is this enough? [JSON.stringify does more](https://262.ecma-international.org/12.0/#sec-quotejsonstring).

It might be worth doing JSON.stringify first and then escape the extras.

Reply:
I could but that would create a string wrapped in extra `\"` quotes which I'd have to strip out. any downside to adding the missing characters JSON.stringify escapes?

Reply:
decided to go the stringify route